core/market/decentralized/src/protocol/negotiation/provider.rs

```diff
@@ -74,7 +88,7 @@ impl NegotiationApi {
         owner: NodeId,
     ) -> Result<(), ProposalError> {
         let msg = ProposalRejected {
-            proposal_id: proposal_id.to_string(),
+            proposal_id: ProposalId::from_str(&proposal_id).unwrap(),
```

Comment:
to consider: translate error instead of unwrapping

Reply:
Hmm I will take ProposalId in parameter rather.
This is refactor artifact ;)
This function is not implemented yet fully, but it seems like implemented, so I will change it to not cause confusion.